source/common/tracing/http_tracer_manager_impl.cc

```diff
@@ -18,11 +18,19 @@ HttpTracerManagerImpl::getOrCreateHttpTracer(const envoy::config::trace::v3::Tra
   const auto cache_key = MessageUtil::hash(*config);
   const auto it = http_tracers_.find(cache_key);
   if (it != http_tracers_.end()) {
-    return it->second;
+    auto http_tracer = it->second.lock();
+    if (http_tracer) { // HttpTracer might have been released since it's a weak reference
+      return http_tracer;
+    } else {
+      http_tracers_.erase(it);
+    }
   }
 
-  // Initialize tracing driver.
-  ENVOY_LOG(info, "instantiating tracing driver: {}", config->name());
+  // Free memory held by expired weak references.
```

Comment:
Can you add a bit of a larger comment around why we don't think this will be a perf issue?

Reply:
added a detailed comment